### PR TITLE
MeshConfig : update comments for types

### PR DESCRIPTION
**Pull request by snehachhabria**
**Description**:

Updates the comments for the API types of MeshConfig

Part of #3040

Signed-off-by: Sneha Chhabria <snchh@microsoft.com>

**Affected area**:

- New Functionality      [ ]
- Documentation          [ ]
- Install                [ ]
- Control Plane          [ ]
- CLI Tool               [ ]
- Certificate Management [ ]
- Networking             [ ]
- Metrics                [ ]
- SMI Policy             [ ]
- Security               [ ]
- Tests                  [ ]
- CI System              [ ]
- Demo                   [ ]
- Performance            [ ]
- Other                  [ ]

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution? `no`

### CODE REVIEW

File: pkg/apis/config/v1alpha1/mesh_config.go [hunk 1]
```diff
@@ -5,66 +5,114 @@ import (
...
+	// Spec is the MeshConfig specification
+	// +optional
 	Spec MeshConfigSpec `json:"spec,omitempty" yaml:"spec,omitempty"`
 }
 
 // MeshConfigSpec is the spec for OSM's configuration
 type MeshConfigSpec struct {
-	Sidecar       SidecarSpec       `json:"sidecar,omitempty"`
-	Traffic       TrafficSpec       `json:"traffic,omitempty"`
+	// Sidecar defines OSM's sidecar configuration
```
Comment (shashankram) [inline pkg/apis/config/v1alpha1/mesh_config.go:27]:
```suggestion
	// Sidecar defines the configurations for the proxy sidecar in a mesh
```

File: pkg/apis/config/v1alpha1/mesh_config.go [hunk 2]
```diff
@@ -5,66 +5,114 @@ import (
...
+	Sidecar SidecarSpec `json:"sidecar,omitempty"`
+
+	// Traffic defines OSM's traffic management configuration
```
Comment (shashankram) [inline pkg/apis/config/v1alpha1/mesh_config.go:30]:
```suggestion
	// Traffic defines the traffic management configurations for a mesh instance
```

File: pkg/apis/config/v1alpha1/mesh_config.go [hunk 3]
```diff
@@ -5,66 +5,114 @@ import (
...
+	Traffic TrafficSpec `json:"traffic,omitempty"`
+
+	// Observalility defines OSM's observability related configuration
```
Comment (shashankram) [inline pkg/apis/config/v1alpha1/mesh_config.go:33]:
```suggestion
	// Observalility defines the observability configurations for a mesh instance
```

File: pkg/apis/config/v1alpha1/mesh_config.go [hunk 4]
```diff
@@ -5,66 +5,114 @@ import (
...
 	Observability ObservabilitySpec `json:"observability,omitempty"`
-	Certificate   CertificateSpec   `json:"certificate,omitempty"`
+
+	// Certificate defines OSM's certificate management configuration
```
Comment (shashankram) [inline pkg/apis/config/v1alpha1/mesh_config.go:36]:
```suggestion
	// Certificate defines the certificate management configurations for a mesh instance
```

File: pkg/apis/config/v1alpha1/mesh_config.go [hunk 5]
```diff
@@ -5,66 +5,114 @@ import (
...
+	Certificate CertificateSpec `json:"certificate,omitempty"`
 }
 
-// SidecarSpec is the spec for OSM's sidecar configuration
+// SidecarSpec is the type used represent the specifications for OSM's sidecar configuration
```
Comment (shashankram) [inline pkg/apis/config/v1alpha1/mesh_config.go:40]:
```suggestion
// SidecarSpec is the type used to represent the specifications for the proxy sidecar
```

File: pkg/apis/config/v1alpha1/mesh_config.go [hunk 6]
```diff
@@ -5,66 +5,114 @@ import (
...
 type SidecarSpec struct {
-	EnablePrivilegedInitContainer bool                        `json:"enablePrivilegedInitContainer,omitempty"`
-	LogLevel                      string                      `json:"logLevel,omitempty"`
-	EnvoyImage                    string                      `json:"envoyImage,omitempty"`
-	InitContainerImage            string                      `json:"initContainerImage,omitempty"`
-	MaxDataPlaneConnections       int                         `json:"maxDataPlaneConnections,omitempty"`
-	ConfigResyncInterval          string                      `json:"configResyncInterval,omitempty"`
-	Resources                     corev1.ResourceRequirements `json:"resources,omitempty"`
+	// EnablePrivilegedInitContainer indicates whether the init container for pods in the mesh should be privileged
```
Comment (shashankram) [inline pkg/apis/config/v1alpha1/mesh_config.go:42]:
```suggestion
	// EnablePrivilegedInitContainer defines a boolean indicating whether the init container for a meshed pod should run as privileged.
```

File: pkg/apis/config/v1alpha1/mesh_config.go [hunk 7]
```diff
@@ -5,66 +5,114 @@ import (
...
+	EnablePrivilegedInitContainer bool `json:"enablePrivilegedInitContainer,omitempty"`
+
+	// LogLevel defines the level of logs collected from envoy
```
Comment (shashankram) [inline pkg/apis/config/v1alpha1/mesh_config.go:45]:
```suggestion
	// LogLevel defines the logging level for the sidecar's logs.
```

File: pkg/apis/config/v1alpha1/mesh_config.go [hunk 8]
```diff
@@ -5,66 +5,114 @@ import (
...
+	LogLevel string `json:"logLevel,omitempty"`
+
+	// EnvoyImage defines the image for envoy sidecar
```
Comment (shashankram) [inline pkg/apis/config/v1alpha1/mesh_config.go:48]:
```suggestion
	// EnvoyImage defines the container image used for the Envoy proxy sidecar.
```

File: pkg/apis/config/v1alpha1/mesh_config.go [hunk 9]
```diff
@@ -5,66 +5,114 @@ import (
...
+	EnvoyImage string `json:"envoyImage,omitempty"`
+
+	// InitContainerImage defines the image for OSM's init container
+	InitContainerImage string `json:"initContainerImage,omitempty"`
+
+	// MaxDataPlaneConnections defines the max allowed data plane sidecar connections
+	MaxDataPlaneConnections int `json:"maxDataPlaneConnections,omitempty"`
+
+	// ConfigResyncInterval defines the resync interval for regular proxy broadcast updates
```
Comment (shashankram) [inline pkg/apis/config/v1alpha1/mesh_config.go:57]:
```suggestion
	// ConfigResyncInterval defines the resync interval for regular proxy broadcast updates.
```

File: pkg/apis/config/v1alpha1/mesh_config.go [hunk 10]
```diff
@@ -5,66 +5,114 @@ import (
...
+	ConfigResyncInterval string `json:"configResyncInterval,omitempty"`
+
+	// Resources defines the compute resources for envoy sidecar
```
Comment (shashankram) [inline pkg/apis/config/v1alpha1/mesh_config.go:60]:
```suggestion
	// Resources defines the compute resources for the sidecar.
```

File: pkg/apis/config/v1alpha1/mesh_config.go [hunk 11]
```diff
@@ -5,66 +5,114 @@ import (
...
 
-// TrafficSpec is the spec for OSM's traffic management configuration
+// TrafficSpec is the type used to represent OSM's traffic management configuration
 type TrafficSpec struct {
-	EnableEgress                      bool     `json:"enableEgress,omitempty"`
-	OutboundIPRangeExclusionList      []string `json:"outboundIPRangeExclusionList,omitempty"`
-	OutboundPortExclusionList         []int    `json:"outboundPortExclusionList,omitempty"`
-	UseHTTPSIngress                   bool     `json:"useHTTPSIngress,omitempty"`
-	EnablePermissiveTrafficPolicyMode bool     `json:"enablePermissiveTrafficPolicyMode,omitempty"`
+	// EnableEgress defines if egress is enabled on the mesh
```
Comment (shashankram) [inline pkg/apis/config/v1alpha1/mesh_config.go:66]:
```suggestion
	// EnableEgress defines a boolean indicating if mesh-wide Egress is enabled.
```

File: pkg/apis/config/v1alpha1/mesh_config.go [hunk 12]
```diff
@@ -5,66 +5,114 @@ import (
...
+	EnableEgress bool `json:"enableEgress,omitempty"`
+
+	// OutboundIPRangeExclusionList defines a global list of IP address ranges to exclude from outbound traffic interception by the sidecar proxy
```
Comment (shashankram) [inline pkg/apis/config/v1alpha1/mesh_config.go:69]:
```suggestion
	// OutboundIPRangeExclusionList defines a global list of IP address ranges to exclude from outbound traffic interception by the sidecar proxy.
```

File: pkg/apis/config/v1alpha1/mesh_config.go [hunk 13]
```diff
@@ -5,66 +5,114 @@ import (
...
+	OutboundIPRangeExclusionList []string `json:"outboundIPRangeExclusionList,omitempty"`
+
+	// OutboundPortExclusionList defines a global list of ports to exclude from outbound traffic interception by the sidecar proxy
```
Comment (shashankram) [inline pkg/apis/config/v1alpha1/mesh_config.go:72]:
```suggestion
	// OutboundPortExclusionList defines a global list of ports to exclude from outbound traffic interception by the sidecar proxy.
```

File: pkg/apis/config/v1alpha1/mesh_config.go [hunk 14]
```diff
@@ -5,66 +5,114 @@ import (
...
+	OutboundPortExclusionList []int `json:"outboundPortExclusionList,omitempty"`
+
+	// UseHTTPSIngress defines if HTTPS Ingress is enabled on the mesh
```
Comment (shashankram) [inline pkg/apis/config/v1alpha1/mesh_config.go:75]:
```suggestion
	// UseHTTPSIngress defines a boolean indicating if HTTPS Ingress is enabled globally in the mesh.
```

File: pkg/apis/config/v1alpha1/mesh_config.go [hunk 15]
```diff
@@ -5,66 +5,114 @@ import (
...
+	UseHTTPSIngress bool `json:"useHTTPSIngress,omitempty"`
+
+	// EnablePermissiveTrafficPolicyMode defines if policy enforcement is enabled on the mesh
```
Comment (shashankram) [inline pkg/apis/config/v1alpha1/mesh_config.go:78]:
```suggestion
	// EnablePermissiveTrafficPolicyMode defines a boolean indicating if permissive traffic policy mode is enabled mesh-wide.
```

File: pkg/apis/config/v1alpha1/mesh_config.go [hunk 16]
```diff
@@ -5,66 +5,114 @@ import (
...
+	EnablePermissiveTrafficPolicyMode bool `json:"enablePermissiveTrafficPolicyMode,omitempty"`
 }
 
-// ObservabilitySpec is the spec for OSM's observability related configuration
+// ObservabilitySpec is the type to represent OSM's observability related configuration
```
Comment (shashankram) [inline pkg/apis/config/v1alpha1/mesh_config.go:82]:
```suggestion
// ObservabilitySpec is the type to represent OSM's observability configurations.
```

File: pkg/apis/config/v1alpha1/mesh_config.go [hunk 17]
```diff
@@ -5,66 +5,114 @@ import (
...
 type ObservabilitySpec struct {
-	EnableDebugServer  bool        `json:"enableDebugServer,omitempty"`
-	PrometheusScraping bool        `json:"prometheusScraping,omitempty"`
-	Tracing            TracingSpec `json:"tracing,omitempty"`
+	// EnableDebugServer defines if debug endpoint on the osm-controller pod is enabled
```
Comment (shashankram) [inline pkg/apis/config/v1alpha1/mesh_config.go:84]:
```suggestion
	// EnableDebugServer defines if the debug endpoint on the OSM controller pod is enabled.
```

File: pkg/apis/config/v1alpha1/mesh_config.go [hunk 18]
```diff
@@ -5,66 +5,114 @@ import (
...
+	EnableDebugServer bool `json:"enableDebugServer,omitempty"`
+
+	// PrometheusScraping defines if prometheus scraping is enabled on the mesh
```
Comment (shashankram) [inline pkg/apis/config/v1alpha1/mesh_config.go:87]:
```suggestion
	// PrometheusScraping defines a boolean indicating if sidecars should be configured for Prometheus metrics scraping.
```

File: pkg/apis/config/v1alpha1/mesh_config.go [hunk 19]
```diff
@@ -5,66 +5,114 @@ import (
...
+	PrometheusScraping bool `json:"prometheusScraping,omitempty"`
+
+	// Tracing defines OSM's tracing configuration
```
Comment (shashankram) [inline pkg/apis/config/v1alpha1/mesh_config.go:90]:
```suggestion
	// Tracing defines OSM's tracing configuration.
```

File: pkg/apis/config/v1alpha1/mesh_config.go [hunk 20]
```diff
@@ -5,66 +5,114 @@ import (
...
+	Tracing TracingSpec `json:"tracing,omitempty"`
 }
 
-// TracingSpec is the spec for OSM's tracing configuration
+// TracingSpec is the type to represent OSM's tracing configuration
```
Comment (shashankram) [inline pkg/apis/config/v1alpha1/mesh_config.go:94]:
```suggestion
// TracingSpec is the type to represent OSM's tracing configuration.
```

File: pkg/apis/config/v1alpha1/mesh_config.go [hunk 21]
```diff
@@ -5,66 +5,114 @@ import (
...
 type TracingSpec struct {
-	Enable   bool   `json:"enable,omitempty"`
-	Port     int16  `json:"port,omitempty"`
-	Address  string `json:"address,omitempty"`
+	// Enable defines if Jaeger tracing is enabled on the mesh
```
Comment (shashankram) [inline pkg/apis/config/v1alpha1/mesh_config.go:96]:
```suggestion
	// Enable defines a boolean indicating if the sidecars are enabled for tracing.
```

File: pkg/apis/config/v1alpha1/mesh_config.go [hunk 22]
```diff
@@ -5,66 +5,114 @@ import (
...
+	Enable bool `json:"enable,omitempty"`
+
+	// Port defines the port on which tracing is enabled
```
Comment (shashankram) [inline pkg/apis/config/v1alpha1/mesh_config.go:99]:
```suggestion
	// Port defines the tracing collector's port.
```

File: pkg/apis/config/v1alpha1/mesh_config.go [hunk 23]
```diff
@@ -5,66 +5,114 @@ import (
...
+	Port int16 `json:"port,omitempty"`
+
+	// Address defines the address of Jaeger tracing deployment
```
Comment (shashankram) [inline pkg/apis/config/v1alpha1/mesh_config.go:102]:
```suggestion
	// Address defines the tracing collector's Hostname.
```

File: pkg/apis/config/v1alpha1/mesh_config.go [hunk 24]
```diff
@@ -5,66 +5,114 @@ import (
...
+	Address string `json:"address,omitempty"`
+
+	// Endpoint defines the endpoint for tracing data
```
Comment (shashankram) [inline pkg/apis/config/v1alpha1/mesh_config.go:105]:
```suggestion
	// Endpoint defines the API endpoint for tracing requests sent to the collector.
```

File: pkg/apis/config/v1alpha1/mesh_config.go [hunk 25]
```diff
@@ -5,66 +5,114 @@ import (
...
 	Endpoint string `json:"endpoint,omitempty"`
 }
 
-// CertificateSpec is the spec for OSM's certificate management configuration
+// CertificateSpec is type to reperesent OSM's certificate management configuration
```
Comment (shashankram) [inline pkg/apis/config/v1alpha1/mesh_config.go:109]:
```suggestion
// CertificateSpec is type to reperesent OSM's certificate management configuration.
```

File: pkg/apis/config/v1alpha1/mesh_config.go [hunk 26]
```diff
@@ -5,66 +5,114 @@ import (
...
 type CertificateSpec struct {
+	// ServiceCertValidityDuration defins the service certificate validity duration
```
Comment (shashankram) [inline pkg/apis/config/v1alpha1/mesh_config.go:111]:
```suggestion
	// ServiceCertValidityDuration defines the service certificate validity duration.
```

**Review by shashankram — CHANGES_REQUESTED** [review]
Thanks for doing this so we can generate API documentation for these.
I have been intentionally verbose with this review because I'd like the API documentation to be consistent in style and wording. I also provided some suggestions to improve clarity.
Should be good to go after these comments are resolved.

**Review by eduser25 — APPROVED** [review]

File: pkg/apis/config/v1alpha1/mesh_config.go [hunk 27]
```diff
@@ -5,66 +5,114 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// MeshConfig is the configuration for the service mesh overall
+// MeshConfig is the type used to configure the mesh
```
Comment (shashankram) [inline pkg/apis/config/v1alpha1/mesh_config.go:8]:
```suggestion
// MeshConfig is the type used to represent the mesh configuration.
```

**Review by shashankram — APPROVED** [review]

**Pull request merged**